modin/config/envvars.py

```diff
@@ -145,10 +145,21 @@ class Memory(EnvironmentVariable, type=int):
 
 class NPartitions(EnvironmentVariable, type=int):
     """
-    How many partitions to use by default
+    How many partitions to use for a Modin DataFrame (along each axis)
     """
 
     varname = "MODIN_NPARTITIONS"
+    # This flag is used to detect whether NPartitions is default value or not
+    _is_default = True
+
+    @classmethod
+    def put(cls, value):
+        cls._is_default = False
+        return super().put(value)
+
+    @classmethod
+    def is_default(cls):
+        return cls._is_default
```

Comment:
it came to my mind that there's probably even more elegant solution:
```suggestion
    @classmethod
    def put_if_default(cls, value):
        """Put specific value if NPartitions wasn't set by user yet"""
        if cls._is_default:
            cls.put(value)
```

and then you do stuff like `NPartitions.put_if_default(len(client.cores()))`

Reply:
Fixed.